storage/factory/pruningStorerFactory.go

```diff
@@ -597,18 +597,34 @@ func (psf *StorageServiceFactory) createTrieEpochRootHashStorerIfNeeded() (stora
 }
 
 func (psf *StorageServiceFactory) createPruningPersister(arg *pruning.StorerArgs) (storage.Storer, error) {
-	if !psf.prefsConfig.FullArchive {
+	isFullArchive := psf.prefsConfig.FullArchive
+	isDBLookupExtenstion := psf.generalConfig.DbLookupExtensions.Enabled
+	if !isFullArchive && !isDBLookupExtenstion {
 		return pruning.NewPruningStorer(arg)
 	}
 
+	numOldActivePersisters := psf.getNumActivePersistersForFullHistoryStorer(isFullArchive, isDBLookupExtenstion)
 	historyArgs := &pruning.FullHistoryStorerArgs{
 		StorerArgs:               arg,
-		NumOfOldActivePersisters: psf.generalConfig.StoragePruning.FullArchiveNumActivePersisters,
+		NumOfOldActivePersisters: numOldActivePersisters,
 	}
 
 	return pruning.NewFullHistoryPruningStorer(historyArgs)
 }
 
+func (psf *StorageServiceFactory) getNumActivePersistersForFullHistoryStorer(isFullArchive bool, isDBLookupExtension bool) uint32 {
+	if isFullArchive && !isDBLookupExtension {
+		return psf.generalConfig.StoragePruning.FullArchiveNumActivePersisters
+	}
+
+	if !isFullArchive && isDBLookupExtension {
+		return psf.generalConfig.DbLookupExtensions.DbLookupMaxActivePersisters
+	}
+
+	log.Warn("node is started with both Full Archive and DB Lookup Extension modes. It will use NumOfOldActivePersisters from full archive's settings")
```

Comment:
as stated, let's change this a little bit:
```GO
if psf.generalConfig.DbLookupExtensions.DbLookupMaxActivePersisters != psf.generalConfig.StoragePruning.FullArchiveNumActivePersisters{
    log.Warn("node is started with both Full Archive and DB Lookup Extension modes and have different values for the number of active persisters. It will use NumOfOldActivePersisters from full archive's settings")
}

return psf.generalConfig.StoragePruning.FullArchiveNumActivePersisters
```

Reply:
done